### ISSUE TITLE
Add support for live update archive manifest

### ISSUE THREAD

**Issue by JCash** [defold/defold#8160]
To further simplify the maintenance of LiveUpdate archive(s), we want to create an easier way of creating and maintaining the archives.

In short, the manifest will allow for:
* Creating one or more live update archives
* Specify what resources will go into which archives

We'll keep the logic as low complexity as possible, with operations like `include` and `exclude` patterns.
The patterns may be collections (e.g. like what we do now, but also other files)
E.g. we may put custom content (e.g. .json files) in an archive.

We can also use other archives as an include/exclude pattern.

In short, it should be possible for the user to completely control which files end up in an archive.

For the editor (perhaps MVP2), we need to be able to see what the outputs archives will contain, in order to ensure complete control for the user.

A more thorough design will follow.

**Comment by SalavatR** [comment]
Hi!
We are working on liveupdate support for our projects, now we can divide builded archive by collections and common files . 
Now it's over 50 archives, and every archive should have dmanifest inside. Maybe it's easier to mount archive with downloaded dmanifest? something like

```
    liveupdate.store_manifest(manifest)
    liveupdate.add_mount(liveupdate.get_current_manifest(), filename, "zip:" .. get_local_path(filename),priority, function(...)
        if cb then
            cb(true)
        end
    end)
```

**Comment by JCash** [comment]
I don't think that's what we want. We want each .zip file to be self-contained. The idea is that the game can load a .zip file from a long time ago (as long as it's supported) or a zip file with new content.

Your current workflow works, but it isn't ideal ofc.
Ideally, the manifest would only describe the files within the zip archive.

This is why I created this ticket, to make it much easier to create and maintain archives.
And then each archive would only contain the info required from the manifest.

**Comment by Morgerion** [comment]
I'd like to revisit this incredibly useful feature, as it will allow you to organize game modding!
Yes, i speak for myself of course, as having modding is critical in our game. :)
Nevertheless, it would be a very very very handy tool.

**Comment by maningame** [comment]
I have the same problems with dmanifest like SalavatR. My game (html5) already have 100+ archives with downloadable content. Each archive contains 226kb liveupdate.game.dmanifest that leads to 14 mb overhead. It will be nice to have an option to load dmanifest only once with first archive.
